core/jvmMain/src/kotlinx/io/buffer/JvmBufferAllocator.kt

```diff
@@ -0,0 +1,12 @@
+package kotlinx.io.buffer
+
+import java.nio.*
+import java.nio.ByteOrder
+
+actual object PlatformBufferAllocator : BufferAllocator {
+    override fun allocate(size: Int): Buffer {
+        return Buffer(ByteBuffer.allocate(size).order(ByteOrder.BIG_ENDIAN))
```

Comment:
Do we plan to have a switch between `allocate/allocateDirect`?

Reply:
We probably will guard it with system property, but allocators are (still) pluggable